services/ai-scribe/gemini_service.py

```
@@ -1,133 +1,84 @@
-            if hasattr(e, 'response') and hasattr(e.response, 'text'):
-                logger.error(f"Gemini API Error Response: {e.response.text}")
-            raise  # Re-raise the exception to be handled by the caller
+    async def call_gemini_api(self, prompt_text, model_name=None, image_base64=None, image_mime_type=None):
+        try:
+            model = genai.GenerativeModel(model_name or config.GEMINI_DEFAULT_MODEL)
+            response = await model.generate_content_async(self._content(prompt_text, image_base64, image_mime_type))
+            text, _ = _candidate_text(response)
+            if not text.strip():
+                raise GenerationError("Generation returned no usable text.")
+            return text
+        except Exception as exc:
```

Ruff (0.16.3)

[warning] Do not catch blind exception: `Exception`

(BLE001)